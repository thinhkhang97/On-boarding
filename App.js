/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, { Component } from 'react';
import Onboarding from 'features/Onboarding';
import { DATA_SAMPLE, THEME } from 'asset/dataSample';

type Props = {};
export default class App extends Component<Props> {
  render() {
    return <Onboarding userData={DATA_SAMPLE} theme={THEME} />;
  }
}
