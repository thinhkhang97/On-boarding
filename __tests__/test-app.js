import * as React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';

test('Render App correctly', ()=>{
    const testApp = renderer.create(<App/>).toJSON();
    expect(testApp).toMatchSnapshot();
})