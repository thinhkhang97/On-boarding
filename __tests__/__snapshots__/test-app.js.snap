// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        horizontal={true}
        onScroll={[Function]}
        pagingEnabled={true}
        showsHorizontalScrollIndicator={false}
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "height": "100%",
                  "width": 750,
                },
                Object {},
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "75%",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  style={
                    Array [
                      false,
                      Object {},
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "white",
                        "borderColor": "green",
                        "borderRadius": 127,
                        "borderWidth": 54,
                        "height": 254,
                        "marginBottom": 48,
                        "overflow": "hidden",
                        "padding": 15,
                        "width": 254,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR2XnXChz496P_XvBFy4e7xp59-y7seeneFBFIYDFnVUQz8U9Fi",
                        }
                      }
                      style={
                        Array [
                          Object {
                            "borderRadius": 58,
                            "height": 116,
                            "width": 116,
                          },
                          Object {
                            "flex": 1,
                            "width": "100%",
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "space-around",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontSize": 32,
                          "fontWeight": "bold",
                        },
                        Array [
                          Object {
                            "color": "green",
                          },
                          Object {
                            "textAlign": "center",
                          },
                        ],
                      ]
                    }
                  >
                    Welcome, Beckham
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#7f8da2",
                          "fontSize": 16,
                        },
                        Object {
                          "paddingLeft": 39,
                          "paddingRight": 36,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Before using Grove, we would like to show you a quick recap of over 3 years working for the company.
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "height": "100%",
                  "width": 750,
                },
                Object {},
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "75%",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "paddingTop": 30,
                        "width": "100%",
                      },
                      Object {},
                    ]
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={
                      Object {
                        "testUri": "../../../asset/images/view2.png",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "height": "100%",
                          "width": "auto",
                        },
                        Object {},
                      ]
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "space-around",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontSize": 32,
                          "fontWeight": "bold",
                        },
                        Array [
                          Object {
                            "color": "green",
                          },
                          Object {
                            "textAlign": "center",
                          },
                        ],
                      ]
                    }
                  >
                    8320 Hrs
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#7f8da2",
                          "fontSize": 16,
                        },
                        Object {
                          "paddingLeft": 39,
                          "paddingRight": 36,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Approx. working hours at VISA Global. Thats ten times Elon Musk believes that his ship will be able to fly to Mars
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "height": "100%",
                  "width": 750,
                },
                Object {},
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "75%",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "paddingTop": 30,
                        "width": "100%",
                      },
                      Object {},
                    ]
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={
                      Object {
                        "testUri": "../../../asset/images/view3.png",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "height": "100%",
                          "width": "auto",
                        },
                        Object {},
                      ]
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "space-around",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "display": "flex",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "display": "flex",
                          "marginLeft": 15,
                          "marginRight": 15,
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "black",
                              "fontSize": 32,
                              "fontWeight": "bold",
                            },
                            Object {
                              "color": "green",
                            },
                          ]
                        }
                      >
                        3
                      </Text>
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "black",
                              "fontSize": 20,
                            },
                            Object {
                              "color": "green",
                            },
                          ]
                        }
                      >
                        Projects
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "display": "flex",
                          "marginLeft": 15,
                          "marginRight": 15,
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "black",
                              "fontSize": 32,
                              "fontWeight": "bold",
                            },
                            Object {
                              "color": "green",
                            },
                          ]
                        }
                      >
                        90
                      </Text>
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "black",
                              "fontSize": 20,
                            },
                            Object {
                              "color": "green",
                            },
                          ]
                        }
                      >
                        Connections
                      </Text>
                    </View>
                  </View>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#7f8da2",
                          "fontSize": 16,
                        },
                        Object {
                          "paddingLeft": 39,
                          "paddingRight": 36,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Nam dapibus nisl vitae elit fringilla rutrum. Aenean sollicitudin, erat a elementum rutrum, neque sem.
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "height": "100%",
                  "width": 750,
                },
                Object {},
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "75%",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "paddingTop": 30,
                        "width": "100%",
                      },
                      Object {},
                    ]
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={
                      Object {
                        "testUri": "../../../asset/images/view4.png",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "height": "100%",
                          "width": "auto",
                        },
                        Object {},
                      ]
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "space-around",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontSize": 32,
                          "fontWeight": "bold",
                        },
                        Array [
                          Object {
                            "color": "green",
                          },
                          Object {
                            "textAlign": "center",
                          },
                        ],
                      ]
                    }
                  >
                    Multiple promotions
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#7f8da2",
                          "fontSize": 16,
                        },
                        Object {
                          "paddingLeft": 39,
                          "paddingRight": 36,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    You’ve  been promoted 3 times. Keep climbing, aim for the summit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "height": 100,
            "justifyContent": "center",
            "marginTop": 40,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": 56,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "green",
                "borderRadius": 4,
                "height": 8,
                "marginBottom": 8,
                "marginTop": 8,
                "opacity": 1,
                "width": 8,
              }
            }
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#33314057",
                "borderRadius": 4,
                "height": 8,
                "marginBottom": 8,
                "marginTop": 8,
                "opacity": 1,
                "width": 8,
              }
            }
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#33314057",
                "borderRadius": 4,
                "height": 8,
                "marginBottom": 8,
                "marginTop": 8,
                "opacity": 1,
                "width": 8,
              }
            }
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#33314057",
                "borderRadius": 4,
                "height": 8,
                "marginBottom": 8,
                "marginTop": 8,
                "opacity": 1,
                "width": 8,
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "green",
              "borderRadius": 22,
              "display": "flex",
              "height": 44,
              "justifyContent": "center",
              "marginBottom": 8,
              "marginTop": 8,
              "opacity": 1,
              "paddingLeft": 10,
              "paddingRight": 10,
              "shadowColor": "#731154",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowRadius": 9,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
              }
            }
          >
            Next
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
